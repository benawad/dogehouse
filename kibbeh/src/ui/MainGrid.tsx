import React from "react";
import { useScreenType } from "../shared-hooks/useScreenType";

interface DashboardGridProps {
  className?: string;
}

export const MainInnerGrid: React.FC<DashboardGridProps> = ({
  children,
  className = "",
}) => {
  const screenType = useScreenType();

  let gridTemplateColumns = "235px 640px 325px";
  let myClassName = ``;

  if (screenType === "2-cols") {
    gridTemplateColumns = "60px 640px 325px";
  } else if (screenType === "1-cols") {
    gridTemplateColumns = "60px 640px";
  } else if (screenType === "fullscreen") {
    myClassName = "w-full px-3";
    gridTemplateColumns = "1fr";
  }

  return (
    <div
      id="main"
      className={`relative ${myClassName} ${className}`}
      style={{
        display: screenType === "fullscreen" ? "flex" : "grid",
        gridTemplateColumns,
        columnGap: 60,
      }}
    >
      {children}
    </div>
  );
};

export const MainGrid: React.FC<DashboardGridProps> = ({ children }) => {
  return (
    <div
      className={`flex justify-center w-full min-h-screen bg-primary-900`}
      data-testid="main-grid"
    >
      <MainInnerGrid>{children}</MainInnerGrid>
    </div>
  );
};
