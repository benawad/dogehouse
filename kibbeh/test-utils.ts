import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import "mutationobserver-shim";

const customRender = (ui: any, options = {}) =>
  render(ui, {
    ...options,
  });

export * from "@testing-library/react";

export { customRender as render };
