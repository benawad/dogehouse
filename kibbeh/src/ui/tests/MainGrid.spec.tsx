import React from "react";

import { render } from "../../../test-utils";
import { MainGrid } from "../MainGrid";

describe("MainGrid", () => {
  it("should render correctly", () => {
    const { getByTestId, getByText } = render(<MainGrid>Child</MainGrid>);
    const component = getByTestId("main-grid");
    const child = getByText("Child");

    expect(component).toBeVisible();
    expect(child).toBeVisible();
  });

  it("should match snapshot", () => {
    const { getByTestId } = render(<MainGrid>Child</MainGrid>);
    const component = getByTestId("main-grid");

    expect(component).toMatchSnapshot();
  });
});
