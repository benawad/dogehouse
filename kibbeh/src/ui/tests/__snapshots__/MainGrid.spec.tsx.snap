// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainGrid should match snapshot 1`] = `
<div
  class="flex justify-center w-full min-h-screen bg-primary-900"
  data-testid="main-grid"
>
  <div
    class="relative w-full px-3 "
    id="main"
    style="display: flex; grid-template-columns: 1fr; column-gap: 60px;"
  >
    Child
  </div>
</div>
`;
